x-pack/plugins/ml/public/application/jobs/new_job/common/job_validator/validators.ts

```diff
@@ -39,39 +63,119 @@ export function isCardinalityModelPlotHigh(
+        return {
+          jobCreator,
+          analysisConfigString: JSON.stringify(jobCreator.jobConfig.analysis_config, null, 2),
+        };
+      }),
+      distinctUntilChanged((prev, curr) => {
+        return prev.analysisConfigString === curr.analysisConfigString;
+      }),
+      switchMap(({ jobCreator }) => {
+        // Perform a cardinality check only with enabled model plot.
+        return ml
+          .validateCardinality$({
+            ...jobCreator.jobConfig,
+            datafeed_config: jobCreator.datafeedConfig,
+          } as CombinedJob)
+          .pipe(
+            map((validationResults) => {
+              for (const validationResult of validationResults) {
+                if (isCardinalityModelPlotHigh(validationResult)) {
+                  return {
+                    highCardinality: {
+                      value: validationResult.modelPlotCardinality,
+                      severity: VALIDATOR_SEVERITY.WARNING,
+                    },
+                  };
+                }
+              }
+              return null;
+            })
+          );
+      }),
+      startWith(null)
+    ),
+  ]).pipe(
+    map(([isModelPlotEnabled, cardinalityValidationResult]) => {
+      return isModelPlotEnabled ? cardinalityValidationResult : null;
+    })
+  );
+}
+
+export function jobIdValidator(jobCreator$: Subject<JobCreator>): Observable<JobExistsResult> {
   return jobCreator$.pipe(
-    // Perform a cardinality check only with enabled model plot.
-    filter((jobCreator) => {
-      return jobCreator?.modelPlot;
+    map((jobCreator) => {
+      return {
+        jobCreator,
+        jobId: jobCreator.jobId,
+      };
+    }),
+    // No need to perform an API call if the analysis configuration hasn't been changed
+    distinctUntilChanged((prev, curr) => {
+      return prev.jobId === curr.jobId;
+    }),
+    switchMap(({ jobId }) => {
+      return ml.jobs.jobsExist$([jobId], true);
     }),
+    map((jobExistsResults) => {
+      const jobs = Object.values(jobExistsResults);
+      if (jobs[0] && jobs[0].exists === true) {
+        return {
+          jobIdExists: {
+            valid: false,
+            message: jobExistsErrorMessage,
+          },
+        };
+      }
+      return {
+        jobIdExists: {
+          valid: true,
+        },
+      };
+    })
+  );
+}
+
+export function groupIdsValidator(jobCreator$: Subject<JobCreator>): Observable<GroupsExistResult> {
+  return jobCreator$.pipe(
     map((jobCreator) => {
       return {
         jobCreator,
-        analysisConfigString: JSON.stringify(jobCreator.jobConfig.analysis_config),
+        groups: jobCreator.groups,
       };
     }),
     // No need to perform an API call if the analysis configuration hasn't been changed
     distinctUntilChanged((prev, curr) => {
-      return prev.analysisConfigString === curr.analysisConfigString;
+      return JSON.stringify(prev.groups) === JSON.stringify(curr.groups);
```

Comment:
we can get rid of the `map` operator before this check

Reply:
`groups` is also a getter function on the `JobCreator` class, so this initial `map` for `groups` is needed here.